manufacturer-toolkit/src/main/java/org/sdo/sct/mt/OnDieSignatureValidator.java

```diff
@@ -0,0 +1,183 @@
+import org.bouncycastle.cert.X509CertificateHolder;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+
+@Service
+public class OnDieSignatureValidator {
+
+  private OnDieCache onDieCache;
+  private boolean checkRevocations = true;
+
+  /**
+   * Constructor.
+   */
+  @Autowired
+  public OnDieSignatureValidator(
+      OnDieCache onDieCache,
+      @Value("${sdo.ondiecache.revocations:true}") final boolean checkRevocations) {
+    this.onDieCache = onDieCache;
+    this.checkRevocations = checkRevocations;
+  }
+
+  /**
+   * Performs a validation of the given signature with the public key
+   * extracted from the given cert chain.
+   *
+   * @param certChain certChain
+   * @param signedData signedData
+   * @param signature signature
+   * @return boolean indicating if signature is valid.
+   * @throws CertificateException when error.
+   */
+  public boolean validate(CertPath certChain,
+                          byte[] signedData,
+                          byte[] signature) throws CertificateException {
+
+    List<Certificate> certificateList = (List<Certificate>) certChain.getCertificates();
+
+    // Check revocations first.
+    if (checkRevocations && !checkRevocations(certificateList)) {
+      return false;
+    }
+
+    try {
+      // check minimum length (taskinfo + R + S)
+      if (signature.length < (36 + 48 + 48)) {
+        return false;
+      }
+      byte[] taskInfo = Arrays.copyOfRange(signature, 0, 36);
+
+      // adjust the signed data
+      // data-to-verify format is: [ task-info | nonce (optional) | data ]
+      // First 36 bytes of signature is the taskinfo. This value must be prepended
+      // to the signed data
+      ByteArrayOutputStream adjSignedData = new ByteArrayOutputStream();
+      adjSignedData.write(Arrays.copyOfRange(signature, 0, taskInfo.length));
+      adjSignedData.write(signedData);
+
+      byte[] adjSignature = convertSignature(signature, taskInfo);
+      PublicKey publicKey = certificateList.get(0).getPublicKey();
+
+      Signature sig = Signature.getInstance("SHA384withECDSA");
+      sig.initVerify(publicKey);
+      sig.update(adjSignedData.toByteArray());
+      return sig.verify(adjSignature);
+    } catch (Exception ex) {
+      return false;
+    }
+  }
+
+  private static byte[] convertSignature(byte[] signature, byte[] taskInfo)
+      throws IllegalArgumentException, IOException {
+    if (taskInfo.length != 36) {
+      throw new IllegalArgumentException("taskinfo length is incorrect: " + taskInfo.length);
+    }
+
+    // Format for signature should be as follows:
+    // 0x30 b1 0x02 b2 (vr) 0x02 b3 (vs)
+    // The b1 = length of remaining bytes,
+    // b2 = length of R value (vr), b3 = length of S value (vs)
+    byte[] rvalue = Arrays.copyOfRange(signature, taskInfo.length, taskInfo.length + 48);
```

Comment:
More details about the hard-coded numbers 48/49/0x30 etc .. Better to define a variable and provide some comments about what they do.